save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/s3/S3OperationsProperties.kt

```diff
@@ -0,0 +1,66 @@
+package com.saveourtool.save.s3
+
+import reactor.core.scheduler.Schedulers
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import java.net.URI
+import java.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.toJavaDuration
+
+/**
+ * @property endpoint S3 endpoint (URI)
+ * @property bucketName bucket name for all S3 storages
+ * @property prefix a common prefix for all S3 storages
+ * @property credentials credentials to S3
+ * @property httpClient configuration for http client to S3
+ * @property async configuration for async operations
+ */
+data class S3OperationsProperties(
+    val endpoint: URI,
+    val bucketName: String,
+    val prefix: String = "",
+    val credentials: CredentialsProperties,
+    val httpClient: HttpClientProperties = HttpClientProperties(),
+    val async: AsyncProperties = AsyncProperties(),
+) {
+    /**
+     * @property accessKeyId [AwsCredentials.accessKeyId]
+     * @property secretAccessKey [AwsCredentials.secretAccessKey]
+     */
+    data class CredentialsProperties(
+        val accessKeyId: String,
+        val secretAccessKey: String,
+    ) {
+        /**
+         * @return [AwsCredentials] created from this object
+         */
+        fun toAwsCredentials(): AwsCredentials = AwsBasicCredentials.create(accessKeyId, secretAccessKey)
+    }
+
+    /**
+     * @property maxConcurrency
+     * @property connectionTimeout
+     * @property connectionAcquisitionTimeout
+     */
+    data class HttpClientProperties(
+        val maxConcurrency: Int = 5,
+        val connectionTimeout: Duration = 30.seconds.toJavaDuration(),
+        val connectionAcquisitionTimeout: Duration = 1.minutes.toJavaDuration(),
+    )
+
+    /**
+     * @property minPoolSize
+     * @property maxPoolSize
+     * @property queueSize
+     * @property ttl
+     */
+    data class AsyncProperties(
+        val minPoolSize: Int = Schedulers.DEFAULT_POOL_SIZE,
+        val maxPoolSize: Int = Schedulers.DEFAULT_BOUNDED_ELASTIC_SIZE,
+        val queueSize: Int = Schedulers.DEFAULT_BOUNDED_ELASTIC_QUEUESIZE,
+        val ttl: Duration = 60.seconds.toJavaDuration(),
+        // reactor.core.scheduler.BoundedElasticScheduler.DEFAULT_TTL_SECONDS
```

Comment:
?

Reply:
it's comment to `ttl` where I took the default value. `diktatFix` moved it to next line. will move on 1 line up

Reply:
will do it in next PR

Reply:
hm, we should catch such situations in diktat, I didn't remember how, but we definitely have logic for inline comments
I suppose, that we shouldn't trigger on such cases. But may be it depends on configuration

Reply:
ok, strange then, but will move to KDoc anyway. It's more proper place for it

Reply:
fixed here d9545ee84223c1a4a3fe8973dc6addc01f12fc80